diff --git a/appl/cmd/mdb.b b/appl/cmd/mdb.b
index 9dd721d..7ab1299 100644
--- a/appl/cmd/mdb.b
+++ b/appl/cmd/mdb.b
@@ -311,8 +311,6 @@ usage()
 	raise "fail:usage";
 }
 
-writeable := 0;
-
 init(nil: ref Draw->Context, argv: list of string)
 {
 	sys = load Sys Sys->PATH;
@@ -330,14 +328,18 @@ init(nil: ref Draw->Context, argv: list of string)
 	dis = load Dis Dis->PATH;
 	dis->init();
 
+	oflags := Sys->OREAD;
 	if (len argv < 2)
 		usage();
 	if (argv != nil)
 		argv = tl argv;
 	if (argv != nil && len hd argv && (hd argv)[0] == '-') {
-		if (hd argv != "-w")
+		if (hd argv == "-w")
+			oflags = Sys->ORDWR;
+		else if (hd argv == "-wo")
+			oflags = Sys->OWRITE;
+		else
 			usage();
-		writeable = 1;
 		argv = tl argv;
 	}
 	if (argv == nil)
@@ -348,9 +350,6 @@ init(nil: ref Draw->Context, argv: list of string)
 	if(argv != nil)
 		cmd = hd argv;
 
-	oflags := Sys->OREAD;
-	if (writeable)
-		oflags = Sys->ORDWR;
 	mfd = sys->open(fname, oflags);
 	if(mfd == nil) {
 		sys->fprint(stderr, "mdb: cannot open %s: %r\n", fname);
